### PR TITLE
Dev/arrow

### PR DESCRIPTION
**Pull request by lmeyerov**
Building block for pivot arrow uploads

### Added
- **js-upload-api**: Apache Arrow upload support
- **js-upload-api**: Privacy settings support

### Change
- **node-api**: Refactor to importing from `js-upload-api` vs previous code clone
- **node-api**: Refactor exports (non-breaking)

### Docs
- **js-upload-api**: Methods point to examples

### Fix
- **js-upload-api**: Throw error if invalid credentials
- **js-upload-api**: Throw error if file upload has an unrecognized format
- **node-api**: Refactor to use bundled js-upload-api

https://www.loom.com/share/817e8d85621444bcb05ec3e60d1102d3

### CODE REVIEW

File: projects/node-api-test/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "author": "Graphistry, Inc.",
   "license": "Apache-2.0",
   "dependencies": {
-    "@graphistry/node-api": "^4.4.1"
+    "@graphistry/node-api": "^4.4.1",
+    "apache-arrow": "^11.0.0"
```
Comment (aucahuasi) [inline projects/node-api-test/package.json:20]: [RAPIDS 23.02](https://github.com/graphistry/graphistry/blob/master/compose/dockerfiles/base/05-nvidia.Dockerfile#L15) is using [Arrow 10.x](https://github.com/rapidsai/cudf/blob/branch-23.02/conda/recipes/libcudf/conda_build_config.yaml#L20)
I don't know if is super critical to keep the same arrow version across all our stack, but oftentimes I found that it's a good idea to do so.

File: projects/node-api/src/Client.ts
Comment (aucahuasi) [inline projects/node-api/src/Client.ts:1]: out of curiosity: why don't we delete the `node-api` folder? Seems `node-api` is just a thin wrapper for `js-upload-api`.

**Review by aucahuasi — COMMENTED** [review]
Thanks Leo, I'll use this branch for my work!

File: projects/node-api-test/package.json (hunk 1 above)
Comment (lmeyerov) [reply projects/node-api-test/package.json:20]: I think we're fine wrt client, b/c format should be stable and this is about a js client -> our python

But once the server is upgraded (tonight) with a fix around categorical uploads, I can play with doing a lower minimum apache-arrow here

Comment (aucahuasi) [reply projects/node-api-test/package.json:20]: Got it!

File: projects/node-api/src/Client.ts
Comment (lmeyerov) [reply projects/node-api/src/Client.ts:1]: yeah i think we're getting there, this was a good first step to that :) 

the main difference so far w/ `js-upload-api` wrt  `client-api` (browser user) vs `node-api` (node) is that a different `fetch()` has to get passed in, the rest has been isomorphic . There might be some more `package.json` & build magic we can do for browser vs server bundling that takes care of that, tho a bit hard in esm...

**Review by exrhizo — COMMENTED** [review]
I have looked through the PR, I'd need to try using it to really understand how this all works. I didn't notice anything immediately.

**Pull request merged**